app/javascript/packages/validated-field/index.js

```diff
@@ -63,8 +69,16 @@ export class ValidatedField extends HTMLElement {
       this.removeChild(this.errorMessage);
       this.errorMessage = null;
     }
+  }
 
-    this.input?.classList.toggle('usa-input--error', !!message);
+  /**
+   * Sets input attributes corresponding to given validity state.
+   *
+   * @param {boolean} isValid Whether input is valid.
+   */
+  setInputIsValid(isValid) {
+    this.input?.classList.toggle('usa-input--error', !isValid);
+    this.input?.setAttribute('aria-invalid', String(!isValid));
```

Comment:
Note that this will behave a bit different from `form-validation.js`, which sets `aria-invalid` on each `input` event. This function is only called...

- On `input`, with `isValid` of `true`, to reset invalid state on input.
- On `invalid` events (form submission), with `isValid` equal to validity state

This revised behavior seems to align with best practices:

>It should not be set to “true” before input validation is performed or the form is submitted.

https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA21